### DISCUSSION TITLE
V2 Process Handling

### DISCUSSION THREAD

**Discussion by JasmineLowen** [elabit/robotmk#349] [General]
I just had a quick look through the repos new v2 implementation in rust and its recent additions, in particular the recent process handling utilities. I was wondering if it would make sense for you to use async rust there instead since manually handling subprocesses could be tricky when the app grows. 

Of course the first thought here would be to use [`tokio`](https://docs.rs/tokio/latest/tokio/index.html) and especially it's [`command` feature](https://docs.rs/tokio/latest/tokio/process/struct.Command.html). The obvious cons are that it will drag in quiet some dependencies with it. There are some more lightweight solutions here as well though if you'd really care about your dependencies like using the [smol](https://github.com/smol-rs/smol) async runtime and then only picking the stuff you really need (like [async-process](https://github.com/smol-rs/async-process))

So: Did you consider these options already? And if so, why did you not choose to use them? (Just curious for that second question)

**Comment by JasmineLowen** [comment]
Just found the article that tickled some of my brain cells when I read the code:

https://notgull.net/why-you-want-async/

Most notable excerpt:

> Likewise, there are a worrying number of Rust programs that contain a bug-ridden version of half of an async runtime.
>
> I call this “poor man’s async”. async/await is a natural pattern for doing multiple things at once; usually, non-async code tends to evolve into something closer and closer to async code, like [carcinisation](https://en.wikipedia.org/wiki/Carcinisation). It’s all over the place in the Rust ecosystem, once you start looking for it.

**Comment by simonmeggle** [comment]
Thanks Robert, I really appreciate your thoughts and feedback! I will discusss this with the development team.
